docs/architecture/index.rst

```diff
@@ -1,25 +1,586 @@
+   concurrently across the available GPUs.
+
+-  *Parallelism within an experiment.* Determined can schedule multiple experiment trials. A
+   hyperparameter search can train multiple trials simultaneously, each on a different GPU.
+
+-  *Parallelism within a trial.* Determined can use multiple GPUs to speed up training of a single
+   trial. Determined coordinates across multiple GPUs on a single machine, or across multiple GPUs
+   on multiple machines, to improve single-trial training performance.
+
+Reproducibility
+===============
+
+Determined supports reproducible machine learning experiments to ensure that Determined experiments
+are deterministic. Rerunning a previous experiment is expected to produce an identical model. This
+ensures that if the model produced from an experiment is ever lost, it can be recovered by rerunning
+the experiment that produced it.
+
+Determined controls and reproduces the following sources of randomness:
+
+-  Hyperparameter sampling decisions.
+-  Initial weights for a given hyperparameter configuration.
+-  Shuffling of training data in a trial.
+-  Dropout or other random layers.
+
+Determined does not currently support controlling non-determinism in floating-point operations.
+
+Configuration
+=============
+
+Determined is a deep learning training platform that simplifies infrastructure management for domain
+experts while enabling configuration-based deep learning functionality. Configuration files control
+the operation and behavior of:
+
+-  Master nodes
+-  Agent nodes
+-  Experiments
+-  Jobs
+
+You can use *configuration templates* to share experiment configurations within an organization.
+
+Provisioning and Deprovisioning
+===============================
+
+A cluster is managed by the master, which provisions and deprovisions agents depending on the
+current volume of experiments on the cluster.
+
+Scheduling
+==========
+
+The master schedules distributed training jobs automatically, ensuring that all of the compute
+resources required for a job are available before the job is launched.
+
+Job queue management is available to the fair share, priority, and Kubernetes preemption schedulers
+and exposes scheduler functionality for visibility and control over scheduling decisions. The *job
+queue* provides information about job ordering and which jobs are queued, which you can manage
+dynamically.
+
+By default, the Kubernetes scheduler does not support gang scheduling or preemption. This can be
+problematic for deep learning workloads that require multiple pods to be scheduled before execution
+starts, such as distributed training. Determined includes built-in support for the `lightweight
+coscheduling plugin
+<https://github.com/kubernetes-sigs/scheduler-plugins/tree/release-1.18/pkg/coscheduling>`__, which
+extends the default Kubernetes scheduler to support gang scheduling. Determined also includes
+support for priority-based scheduling with preemption, although, neither is enabled by default.
```

Comment:
This section doesn't belong yet.  The rest of the document thus far has described agent-based installations.

We almost need an "alternate installations" section for k8s and slurm, and this can be a detail under k8s there.